c/validation/adbc_validation_util.h

```diff
@@ -197,6 +198,22 @@ struct StreamReader {
   }
 };
 
+/// \brief Read an AdbcGetInfoData struct with RAII safety
+struct GetInfoReader {
+  explicit GetInfoReader(struct ArrowArrayView* array_view) : array_view_(array_view) {
```

Comment:
Probably need to do something here to better tie in the lifetime of the `array_view`, though I wasn't sure if that was worth doing as a separate class like this or maybe integrating into the existing `StreamReader` class